codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocket13FrameDecoder.java

```diff
@@ -177,12 +174,13 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in) throws Exception {
             }
 
             state = State.READING_SECOND;
-        case READING_SECOND:
-            if (!in.isReadable()) {
+        }
+        case READING_SECOND: {
+            if (in.readableBytes() <= 0) {
```

Comment:
I think this never can be negative or I am wrong ?
```suggestion
            if (in.readableBytes() == 0) {
```